motionplan/tpspace/ptgGroupFrame_test.go

```diff
@@ -0,0 +1,160 @@
+package tpspace
+
+import (
+	"math"
+	"testing"
+
+	"go.viam.com/test"
+
+	"go.viam.com/rdk/logging"
+	"go.viam.com/rdk/referenceframe"
+	"go.viam.com/rdk/spatialmath"
+)
+
+func TestPtgNegativePartialTransform(t *testing.T) {
+	// diff drive only ptg group frame
+	pFrame, err := NewPTGFrameFromKinematicOptions(
+		"",
+		logging.NewTestLogger(t),
+		1.,
+		2,
+		nil,
+		true,
+		true,
+	)
+	test.That(t, err, test.ShouldBeNil)
+	pf, ok := pFrame.(*ptgGroupFrame)
+	test.That(t, ok, test.ShouldBeTrue)
+
+	partialPose1, err := pf.Transform([]referenceframe.Input{{0}, {math.Pi / 2}, {-30}, {-200}})
+	test.That(t, err, test.ShouldBeNil)
+	partialPose2, err := pf.Transform([]referenceframe.Input{{0}, {math.Pi / 2}, {-40}, {-200}})
+	test.That(t, err, test.ShouldBeNil)
+
+	test.That(
+		t,
+		spatialmath.PoseAlmostEqual(
+			partialPose1,
+			spatialmath.Compose(partialPose2, spatialmath.NewPoseFromOrientation(&spatialmath.OrientationVectorDegrees{OZ: 1, Theta: -10})),
+		),
+		test.ShouldBeTrue,
+	)
+}
+
+func TestInterpolate(t *testing.T) {
+	t.Parallel()
+	// diff drive only ptg group frame
+	pFrame, err := NewPTGFrameFromKinematicOptions(
+		"",
+		logging.NewTestLogger(t),
+		1.,
+		2,
+		nil,
+		false,
+		true,
+	)
+	test.That(t, err, test.ShouldBeNil)
+
+	zeroInput := make([]referenceframe.Input, 4)
+
+	type testCase struct {
+		name      string
+		inputFrom []referenceframe.Input
+		inputTo   []referenceframe.Input
+		expected  []referenceframe.Input
+		amount    float64
+	}
+
+	testCases := []testCase{
```

Comment:
Thanks for laying these tests out this way it makes it really easy to follow the cases and review